Added 1.14.4 and 1.15 support

NgLoader opened the pull request:
Currently finished

NgLoader committed:
Added 1.14.4 R0.1 support

ProgrammerDan commented:
Thank you, will review tonight!

On Wed, Sep 4, 2019 at 5:42 PM NgLoader <notifications@github.com> wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/lishid/Orebfuscator/pull/236
> Commit Summary
>
>    - Added 1.14.4 R0.1 support
>
> File Changes
>
>    - *M* pom.xml
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-0> (1)
>    - *A* v1_14_4_R1/pom.xml
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-1> (46)
>    - *A*
>    v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/BlockInfo.java
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-2> (65)
>    - *A*
>    v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/ChunkCache.java
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-3> (99)
>    - *A*
>    v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NBT.java
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-4> (65)
>    - *A*
>    v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java
>    <https://github.com/lishid/Orebfuscator/pull/236/files#diff-5> (617)
>
> Patch Links:
>
>    - https://github.com/lishid/Orebfuscator/pull/236.patch
>    - https://github.com/lishid/Orebfuscator/pull/236.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lishid/Orebfuscator/pull/236?email_source=notifications&email_token=AADD5DU2U66O2C2AGNURKQ3QIATVVA5CNFSM4ITWZ2V2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HJL3FZA>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADD5DVOGL6V6H2WGDTYZ53QIATVVANCNFSM4ITWZ2VQ>
> .
>

NgLoader committed:
Changed version from plugin to 1.14.4 and fixed false module name

ProgrammerDan commented:
Broad strokes are OK, some details to fine tune -- do you figure you are done with this contribution or do you intend to work on it more? Regardless, thanks for this contribution, will blend with the prior PR, alongside some additional work I intend to begin soon to figure out the longstanding bugs.

ProgrammerDan commented on Plugin/pom.xml:
```diff
@@ -20,7 +20,7 @@
     <dependency>
```
It's likely the POM will need some other changes. The prior PR #235 had the right idea in some places, esp the version increase (although preserve the -SNAPSHOT -- I remove that for releases alone).

ProgrammerDan commented on pom.xml:
```diff
@@ -38,6 +38,7 @@
     <module>v1_12_R1</module>
     <module>v1_13_R1</module>
     <module>v1_13_R2</module>
+    <module>v1_14_4_R1</module>
```
Module should be v1_14_R1 -- so far as I can see there are no API changes, just fix releases, so it's just v1_14_R1.

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/BlockInfo.java:
```diff
@@ -0,0 +1,65 @@
+ *
+ */
+
+package com.lishid.orebfuscator.nms.v1_14_2_R1;
```
package and folder should be v1_14_R1, or at the least, consistent with base folder.

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/ChunkCache.java:
```diff
@@ -0,0 +1,99 @@
+        for (RegionFile regionFile : cachedRegionFiles.values()) {
+            try {
+                if (regionFile != null) {
+                	// This lovely piece of work is due to an NMS change in Spigot 1.13.2 without an R increase.
```
Due to release of obfuscation mapping, I can confidently describe that this extra handler is unnecessary for 1.14, and can be reduce to just the "close" method retrieval and invocation. (line 38987 in the server deobfuscation for 1.14).

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java:
```diff
@@ -0,0 +1,617 @@
+		int chunkZ = z >> 4;
+
+		WorldServer worldServer = ((CraftWorld)world).getHandle();
+		// like in ChunkCache, NMS change without R increment.
```
will have to investigate bukkit here to see which of the following Methods needs to remain

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java:
```diff
@@ -0,0 +1,617 @@
+import net.minecraft.server.v1_14_R1.WorldServer;
+
+public class NmsManager implements INmsManager {
+	private static final int BITS_PER_BLOCK = 14;
```
Not 100% sure but indicators I use, indicate that this bits per block is still 14, so good news there

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java:
```diff
@@ -0,0 +1,617 @@
+				Material.YELLOW_BED,
+				Material.YELLOW_STAINED_GLASS,
+				Material.YELLOW_STAINED_GLASS_PANE,
+				Material.YELLOW_WALL_BANNER
```
Some good expansion here

NgLoader replied:
currently not, sry

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java:
```diff
@@ -0,0 +1,617 @@
+		// 1.13.2 has made this quite a bit different in later builds.
+		TileEntity tileEntity = null;
+		try {
+			Method getTileEntityAt = world.getClass().getMethod("getTileEntityAt", int.class, int.class, int.class);
```
Will have to reference craftbukkit impl to see which one is current for 1.14

ProgrammerDan commented on v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java:
```diff
@@ -0,0 +1,617 @@
+	}
+
+	public boolean isTileEntity(int combinedBlockId) {
+		return Block.getByCombinedId(combinedBlockId).getBlock().isTileEntity();
```
Hopefully the combinedID business still works. Will review.

ProgrammerDan reviewed (changes requested):
Some minor changes, some stuff to double check -- thanks for contributing!

NgLoader commented:
I commit some changes soon 👍

NgLoader committed:
fixed and optimized some funktions

NgLoader commented:
Currently I'm looking into the ArrayIndexOutOfBoundsException error

NgLoader committed:
nms current is now private and only getable with a function

NgLoader committed:
some small fancy changes

NgLoader committed:
Removed debug message

NgLoader committed:
removed unused import

ProgrammerDan commented:
Any updates here, friend?

Chevels commented:
Any news ? Or OFC is definitively abandonned ? :/

ProgrammerDan commented:
Nothing new from the contributor of this PR, and as for me I haven't had much time for any of my projects. Not officially abandoned but I would welcome dedicated help.

SydMontague commented:
I tested the branch locally and it didn't work properly. After some digging I figured out it's because of a protocol change the code doesn't attribute for.
(In detail: light information is not transmitted anymore and a non-air block count field has been introduced)

Based on NgLoader's work I wrote a patch that should make it 1.14.4 compatible, it worked on my local test server but that by no means indicates full compatibility. The patch shouldn't be backward compatible either, so for that some more work is required.

[Orebfuscator114.patch.txt](https://github.com/lishid/Orebfuscator/files/3876734/Orebfuscator114.patch.txt)

ProgrammerDan commented:
Nice, thank you. Will try to do some testing / integration -- would welcome
a full pr as well.

On Thu, Nov 21, 2019 at 4:04 PM SydMontague <notifications@github.com>
wrote:

> I tested the branch locally and it didn't work properly. After some
> digging I figured out it's because of a protocol change the code doesn't
> attribute for.
> (In detail: light information is not transmitted anymore and a non-air
> block count field has been introduced)
>
> Based on NgLoader's work I wrote a patch that should make it 1.14.4
> compatible, it worked on my local test server but that by no means
> indicates full compatibility. The patch shouldn't be backward compatible
> either, so for that some more work is required.
>
> Orebfuscator114.patch.txt
> <https://github.com/lishid/Orebfuscator/files/3876734/Orebfuscator114.patch.txt>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/lishid/Orebfuscator/pull/236?email_source=notifications&email_token=AADD5DRU6EFJUG4MKJSN3YDQU3ZWHA5CNFSM4ITWZ2V2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEE3URIQ#issuecomment-557271202>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADD5DUB7HEQUFOI6JQKWYDQU3ZWHANCNFSM4ITWZ2VQ>
> .
>

NgLoader commented:
I commit some changes soon.
Currently is my pc away for a upgrade.

Thanks @SydMontague

NgLoader commented:
And @Sydmontague you can commit on my frok so you don't need to send a txt file with changes 😅

NgLoader commented:
I moved the project to a new location
[Repo](https://github.com/TeamWuffy/Orebfuscator)

SydMontague committed:
Improved 1.14.4 compatibility

- removed reflection from NmsManager, since it was only for 1.13.2
- added null check to BlockUpdate, since it's nullable
- use blockCount field when using 1.14.4
- don't use lightingArray when using 1.14.4

This should be backwards compatible (tested with 1.13.2).

SydMontague commented:
I created a pull request to your repo and also fixed some more issues. The NmsManager did some obsolete reflection that caused the plugin to not work properly in the Nether. In the process I also found a potential NPE I prevented with a simple null check.

The changes I made should now also be backwards compatible...

NgLoader committed:
Merge pull request #1 from SydMontague/master

Improved 1.14.4 compatibility

NgLoader committed:
Cleanup project

NgLoader committed:
Added CI

NgLoader committed:
Updated CI

NgLoader committed:
Fixed CI error

NgLoader committed:
Fixed CI

NgLoader committed:
Changed artifacts path

team-wuffy committed:
Added settings.xml

team-wuffy committed:
Added release test

team-wuffy committed:
Chnaged CI artifacts and depoly

team-wuffy committed:
Fixed CI Deploy error

team-wuffy committed:
Removed depoly

team-wuffy committed:
Removed useless

team-wuffy committed:
Added realease CI

team-wuffy committed:
Removed contentType from release CI

team-wuffy committed:
Changed artifacts name

team-wuffy committed:
Fixed that no artifacts was found

team-wuffy committed:
Fixed realses CI error

NgLoader committed:
Added new version system

NgLoader committed:
fixed build CI getting false version

NgLoader committed:
fixed release CI

NgLoader committed:
fixed released CI

team-wuffy committed:
testing release ci

NgLoader committed:
testing ci

NgLoader committed:
testing ci

NgLoader committed:
testing ci

NgLoader committed:
changed readme.md

NgLoader committed:
fixed readme.md

team-wuffy committed:
Merge branch 'master' into ci

team-wuffy committed:
Merge pull request #2 from TeamWuffy/ci

Ci

team-wuffy committed:
Update README.md

NgLoader commented:
Its working for me too, you can test the version [here](https://github.com/TeamWuffy/Orebfuscator/releases/latest)

I'm currently finished
Some small updates coming soon

NgLoader committed:
updated readme.md

NgLoader committed:
fix: readme.md

NgLoader commented:
1.15 update coming soon :smile: currently testing works fine

NgLoader committed:
release 4.4.6
added 1.15 support
added tabcomplete

team-wuffy committed:
Merge pull request #3 from TeamWuffy/4.4.6-release

4.4.6 release

Miky88 commented:
rip why closed

NgLoader commented:
The project was moved to Imprex-Development for 1.8-1.15 support.
1.16 ist currently in progress
